src/routes/transactions/mappers/common/twap-order.mapper.ts

```diff
@@ -195,37 +193,110 @@ export class TwapOrderMapper {
+    return activePart ?? null;
+  }
 
-      // Return the status of the last part
-      if (i === orders.length - 1) {
-        return order.status;
+  private async getPartOrders(args: {
+    partsToFetch: Array<GPv2OrderParameters>;
+    chainId: string;
+    safeAddress: `0x${string}`;
+  }): Promise<Array<KnownOrder> | null> {
+    const orders: Array<KnownOrder> = [];
+
+    for (const part of args.partsToFetch) {
+      const partFullAppData = await this.swapsRepository.getFullAppData(
+        args.chainId,
+        part.appData,
+      );
+
+      if (!this.swapAppsHelper.isAppAllowed(partFullAppData)) {
+        throw new Error(
+          `Unsupported App: ${partFullAppData.fullAppData?.appCode}`,
+        );
+      }
+
+      const orderUid = this.gpv2OrderHelper.computeOrderUid({
+        chainId: args.chainId,
+        owner: args.safeAddress,
+        order: part,
+      });
+      const order = await this.swapsRepository
+        .getOrder(args.chainId, orderUid)
+        .catch(() => {
+          this.loggingService.warn(
+            `Error getting orderUid ${orderUid} from SwapsRepository`,
+          );
+        });
+
+      if (!order || order.kind == OrderKind.Unknown) {
+        // Without every order it's not possible to determine executed amounts/fees
+        return null;
       }
 
-      // If the part is complete, continue to the next part
-      if (completeStatuses.includes(order.status)) {
-        continue;
+      if (!this.swapAppsHelper.isAppAllowed(order)) {
+        throw new Error(`Unsupported App: ${order.fullAppData?.appCode}`);
       }
 
-      return order.status;
+      orders.push(order as KnownOrder);
+    }
+
+    return orders;
+  }
+
+  private async getOrderStatus(args: {
+    chainId: string;
+    safeAddress: `0x${string}`;
+    twapParts: Array<GPv2OrderParameters>;
+    partOrders: Array<KnownOrder> | null;
+    activeOrderUid: `0x${string}` | null;
+    executionDate: Date | null;
+  }): Promise<OrderStatus> {
+    if (!args.executionDate) {
+      return OrderStatus.PreSignaturePending;
+    }
+
+    const orderUid =
+      args.activeOrderUid ??
+      this.gpv2OrderHelper.computeOrderUid({
+        chainId: args.chainId,
+        owner: args.safeAddress,
+        // If there is no "active" part, we use the last one to get the status
+        order: args.twapParts.slice(-1)[0],
+      });
+
+    // If we already have the order, there is no need to fetch it again
+    if (args.partOrders) {
+      const order = args.partOrders.find((order) => order.uid === orderUid);
+      if (order) {
+        return order.status;
+      }
     }
 
-    return OrderStatus.Unknown;
+    try {
+      const order = await this.swapsRepository.getOrder(args.chainId, orderUid);
+      // If the active order is fulfilled, the next one may not yet exist
+      // and fetching it would fail so we assume it's open
+      if (args.activeOrderUid && order.status === OrderStatus.Fulfilled) {
```

Comment:
Whenever this order exists we have to return the total twap order status as Open.
Even if the activePart is still open or even if it got cancelled (as it is possible to cancel a single part without cancelling the whole order)